packages/apidom-converter/src/strategies/openapi-3-1-to-openapi-3-0-3/refractor-plugins/reference-summary-description.ts

```diff
@@ -0,0 +1,39 @@
+import { ReferenceElement } from '@swagger-api/apidom-ns-openapi-3-1';
+import { AnnotationElement } from '@swagger-api/apidom-core';
+
+type ReferenceSummaryDescriptionPluginOptions = {
+  annotations: AnnotationElement[];
+};
+
+const referenceSummaryDescriptionRefractorPlugin =
+  ({ annotations }: ReferenceSummaryDescriptionPluginOptions) =>
+  () => {
+    const summaryAnnotation = new AnnotationElement(
+      'The "summary" field of Reference Object is not supported in OpenAPI 3.0.3. It has been removed from the converted document.',
+      { classes: ['warning'] },
+      { code: 'reference-summary' },
+    );
+    const descriptionAnnotation = new AnnotationElement(
+      'The "description" field of Reference Object is not supported in OpenAPI 3.0.3. It has been removed from the converted document.',
+      { classes: ['warning'] },
+      { code: 'reference-description' },
+    );
+
+    return {
+      visitor: {
+        ReferenceElement(element: ReferenceElement) {
+          if (element.hasKey('summary')) {
+            annotations.push(summaryAnnotation);
+            element.remove('summary');
+          }
+          if (element.hasKey('description')) {
+            annotations.push(descriptionAnnotation);
+            element.remove('description');
+          }
+          return undefined;
```

Comment:
No need to return here. the `return` will be implicit and equivalent if we remove this code.

```suggestion
          return undefined;
```